homeassistant/components/switch/broadlink.py

```diff
@@ -136,6 +137,16 @@ def _send_packet(call):
     elif switch_type in SP2_TYPES:
         broadlink_device = broadlink.sp2((ip_addr, 80), mac_addr)
         switches = [BroadlinkSP2Switch(friendly_name, broadlink_device)]
+    elif switch_type in MP1_TYPES:
+        broadlink_device = broadlink.mp1((ip_addr, 80), mac_addr)
+        if switch_type == "mp1-1":
+            switches = [BroadlinkMP1Switch(friendly_name, broadlink_device, 1)]
+        if switch_type == "mp1-2":
+            switches = [BroadlinkMP1Switch(friendly_name, broadlink_device, 2)]
+        if switch_type == "mp1-3":
+            switches = [BroadlinkMP1Switch(friendly_name, broadlink_device, 3)]
+        if switch_type == "mp1-4":
+            switches = [BroadlinkMP1Switch(friendly_name, broadlink_device, 4)]
```

Comment:
Does not all mp1 switches have 4 slots?
Then I think we always should add all 4 slots.

Reply:
yup. I updated the code to add all 4 slots and changed device type to `mp1` only. New config  should be like this
```yaml
- platform: broadlink
  host: 192.168.0.x
  mac: xx:xx:xx:xx:xx:xx
  type: mp1
  friendly_name: 'MP 1'
  slots:
    # friendly name of slots - optional
    # if not set, slot name will be switch's friendly_name + 'slot {slot_index}'. e.g 'MP 1 slot 1'
    slot_1: 'TV slot'
    slot_2: 'Xbox slot'
    slot_3: 'Fan slot'
    slot_4: 'Speaker slot'
```